edgraph/server.go

```diff
@@ -2152,12 +2152,16 @@ func verifyUniqueWithinMutation(qc *queryContext) error {
 	for i := range qc.uniqueVars {
 		gmuIndex, rdfIndex := decodeIndex(i)
 		if len(qc.gmuList[gmuIndex].Set) == 0 {
-			return nil
+			continue
 		}
 		pred1 := qc.gmuList[gmuIndex].Set[rdfIndex]
 		pred1Value := dql.TypeValFrom(pred1.ObjectValue).Value
 		for j := range qc.uniqueVars {
 			gmuIndex2, rdfIndex2 := decodeIndex(j)
+			if len(qc.gmuList[gmuIndex2].Set) == 0 {
```

Comment:
Thank you for the PR! The test fails, so please check. This is not the right solution.